### PR TITLE
chore(content-uploader): Migrated ProgressBar to TS and RTL

### PR DESCRIPTION
**Pull request by greg-in-a-box**

### CODE REVIEW

File: src/elements/content-uploader/ProgressBar.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+};
+
+const ProgressBar = ({ percent }: Props) => {
+    const [currentPercent, setCurrentPercent] = useState(percent || 0);
+
+    useEffect(() => {
+        setCurrentPercent(percent);
+    }, [percent]);
+
+    let controlledPercent = currentPercent > 100 ? 100 : currentPercent;
```
Comment (jstoffan) [inline src/elements/content-uploader/ProgressBar.tsx:15]: Can we use some `Math` operators to clamp these values and avoid multiple ternaries?

```
const clampedPercentage = Math.max(0, Math.min(100, percent))
```

File: src/elements/content-uploader/ProgressBar.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import React, { useState, useEffect } from 'react';
+import './ProgressBar.scss';
+
+type Props = {
+    percent?: number;
+};
+
+const ProgressBar = ({ percent }: Props) => {
+    const [currentPercent, setCurrentPercent] = useState(percent || 0);
```
Comment (jstoffan) [inline src/elements/content-uploader/ProgressBar.tsx:9]: Do we need state at all?

**Review by benjamin-shen — COMMENTED** [review]
are we keeping around `src/elements/content-uploader/ProgressBar.js.flow`?

**Comment by greg-in-a-box** [comment]
> are we keeping around `src/elements/content-uploader/ProgressBar.js.flow`?

yes because the component that is calling that component is still using flow

File: src/elements/content-uploader/__tests__/ProgressBar.test.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+import { render, screen } from '@testing-library/react';
+import ProgressBar from '../ProgressBar';
```
Comment (benjamin-shen) [inline src/elements/content-uploader/__tests__/ProgressBar.test.tsx:3]: which component does this test?

**Review by jstoffan — APPROVED** [review]

File: src/elements/content-uploader/__tests__/ProgressBar.test.tsx (hunk 3 above)
Comment (greg-in-a-box) [reply src/elements/content-uploader/__tests__/ProgressBar.test.tsx:3]: The progressbar when a item is being uplaoded via the contentuploader

**Review by benjamin-shen — APPROVED** [review]

**Pull request merged**